libnavigation-core/src/main/java/com/mapbox/navigation/core/HistoryRecordingStateHandler.kt

```diff
@@ -82,15 +95,19 @@ internal class HistoryRecordingStateHandler(
     ) {
         val newState = NavigationSessionUtils.getNewState(isDriving, hasRoutes)
         if (newState::class != currentState::class) {
-            if (currentState !is NavigationSessionState.Idle) {
-                historyRecordingStateChangeObservers.forEach {
-                    finishRecordingBlock(it, currentState)
+            val oldState = currentState
+            currentState = newState
+            scope.launch {
+                if (oldState !is NavigationSessionState.Idle) {
+                    historyRecordingStateChangeObservers.forEach {
+                        finishRecordingBlock(it, oldState)
+                    }
                 }
+                if (newState !is NavigationSessionState.Idle) {
+                    historyRecordingStateChangeObservers.forEach { it.onShouldStartRecording(newState) }
+                }
+                _sessionIdFlow.emit(newState.sessionId)
```

Comment:
why do we need to expose `sessionId` via metadata if it's already available via `MapboxNavigation.registerNavigationSessionStateObserver` 🤔  ?

Reply:
Because it's another NavigationSessionState that's used for history recording. These states are not the same as the ones passed to `NavigationSessionStateObserver`. Refs: https://github.com/mapbox/mapbox-navigation-android/pull/6074
That's why we need to expose this id separately. Notice that the name is different: `copilotSessionId`. We don't want users to think it's the same.

Reply:
so it may make sense to make them different? If they are not the same by meaning, let's generate them separately.

Reply:
@Guardiola31337 I like this idea, it will make the code less confusing. I think of introducing a new class in the internal package, sth like `CopilotSessionState`, it will for now be the same as `NavigationSessionState` regarding its contents but changes will be needed on Copilot side to make it compile? If it's OK with you I'll cut a ticket and implement it.

Reply:
`NAVAND-750`